pkg/apis/rsyslog/validation/rsyslog_relp.go

```diff
@@ -112,6 +121,17 @@ func validateLoggingRules(loggingRules []rsyslog.LoggingRule, fldPath *field.Pat
 	return allErrs
 }
 
+func validateProgramNames(programNames []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	validatingRegex, _ := regexp.CompilePOSIX(`[[:/]`)
```

Comment:
Can you move this outside the function (right before it on line 123 in a `MustCompile`
```
var programNamesRegex = regexp.MustCompille(`[[:/]`)

func validateProgramNames(programNames []string, fldPath *field.Path) field.ErrorList {
```

We do not need a POSIX regex here.